concert/processes/experiments.py

```diff
@@ -49,12 +80,23 @@ async def run(self):
 
                 self._experiment.walker.descend(await self._get_iteration_name(iteration))
                 exp_run = self._experiment.run()
-                self._experiment.log.info("Sample name: %s" % await self._get_iteration_name(iteration))
+                self._experiment.log.info("Sample name: %s" %
+                                          await self._get_iteration_name(iteration))
+
+                if hasattr(self._experiment, "ready_to_prepare_next_sample"):
```

Comment:
Wouldn't it be worth making something like a signal in pyqt and registering for it here? I can imagine that by _every_ experiment one can define a state when no devices will be touched anymore and for experiments which do not know/care the default implementation can be conservative and set this event at the end of the run. The reason is that I want to move away from these ad-hoc "hasattr" checks, which stem from our "workarounds" back then,

Reply:
instead of a signal, asyncio.Event would work in this case.

Reply:
Yes. I will add this to the base-Experiment and set/reset in the run(), so it will work if the subclass does not implement it.